### PR TITLE
Reflekt models refactoring

### PR DESCRIPTION
**Pull request by nbirillo**
Refactor code, especially a part with models and ReflektModuleAnalysisExtension

### CODE REVIEW

File: reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/analysis/models/Instances.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package org.jetbrains.reflekt.plugin.analysis.models
+
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.reflekt.plugin.analysis.processor.FileID
+import org.jetbrains.reflekt.plugin.analysis.processor.source.instances.*
+
+typealias TypeInstances<T> = MutableList<T>
+typealias ObjectInstances = TypeInstances<KtObjectDeclaration>
+typealias ClassInstances = TypeInstances<KtClass>
+typealias FunctionInstances = TypeInstances<KtNamedFunction>
```
Comment (elena-lyulina) [inline reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/analysis/models/Instances.kt:10]: do we really need typealiases for a mutable list? sometimes it's so tedious to always navigate to the definition just to see it's a list of something...

**Review by elena-lyulina — CHANGES_REQUESTED** [review]
Nice, thanks!

File: reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/analysis/models/Instances.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package org.jetbrains.reflekt.plugin.analysis.models
+
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.reflekt.plugin.analysis.processor.FileID
+import org.jetbrains.reflekt.plugin.analysis.processor.source.instances.*
+
+typealias TypeInstances<T> = MutableList<T>
```
[the remark this answers is not in the document]
Comment (nbirillo) [reply reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/analysis/models/Instances.kt:7]: Sorry, we have a lot of conflicting branches and PRs, if you would like to rename it let's do it after ALL merges. Let's create a new issue

File: reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/analysis/models/Instances.kt (hunk 1 above)
Comment (nbirillo) [reply reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/analysis/models/Instances.kt:10]: Yes, it is used also in ClassOrObjectInstancesProcessor for example. Currently it is a list, but I am not sure tat we keep it in the future

File: reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/analysis/models/Common.kt [hunk 3]
```diff
@@ -1,27 +1,75 @@
...
     ValueArgumentList("VALUE_ARGUMENT_LIST"),
-    ValueParameter("VALUE_PARAMETER"),
     ValueParameterList("VALUE_PARAMETER_LIST"),
 }
 
-data class ReflektContext(
-    var uses: IrReflektUses? = null,
-    var instances: IrReflektInstances? = null
+// TODO: think about a better name
+interface ISizeable {
```
[the remark this answers is not in the document]
Comment (nbirillo) [reply reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/analysis/models/Common.kt:25]: Both of them are strange... I kept Sizeable  and todo

**Comment by nbirillo** [comment]
@elena-lyulina I will merge PR since it does not have global problems. I just want to resolve all conflicts as soon as possible.

**Pull request merged**